pkgs/by-name/pd/pd/package.nix

```diff
@@ -0,0 +1,79 @@
+{ rev ? "a0f8550034958d71cb998bfc1d029bb02ff83320"
+, hash ? "sha256-0He3Ryqy4RxYuR2HbwGHJxegfOuKJ91B/JI+z2AK8wk="
+
+, lib
+
+, stdenv_32bit
+, fetchFromGitHub
+, fetchpatch
+
+, pkgsi686Linux # SDL2 and zlib
+
+, pkg-config
+, python3
+}: let
+  shortrev = builtins.substring 0 9 rev;
+in stdenv_32bit.mkDerivation rec {
+  pname = "pd";
+  version = shortrev;
+
+  meta = with lib; {
+    homepage = "https://github.com/fgsfdsfgs/perfect_dark";
+    description = "A PC port of Perfect Dark based on the decompilation of the Nintendo 64 game";
+    mainProgram = "pd";
+    platforms = [ "i686-linux" "x86_64-linux" ];
+    maintainers = with maintainers; [ PaulGrandperrin ];
+    license = with licenses; [
+      # pd, khrplatform.h, port/fast3d
+      mit
+      # tools/mkrom/gzip
+      gpl3Plus
+    ];
+  };
+
+  src = fetchFromGitHub {
+    owner = "fgsfdsfgs";
+    repo = "perfect_dark";
+    inherit rev hash;
+  };
+
+  buildInputs = with pkgsi686Linux; [
+    SDL2
+    zlib
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    python3
+  ];
+
+  # the project uses git to retrieve version informations but our fetcher deletes the .git
+  # so we replace the commands with the correct data directly
+  postPatch = ''
+    sed -i 's/git rev-parse --short HEAD/echo ${shortrev}/' Makefile.port
+    sed -i 's/git rev-parse --abbrev-ref HEAD/echo port/' Makefile.port
+  '';
+
+  enableParallelBuilding = true;
+
+  hardeningDisable = [ "format" ]; # otherwise fails to build
+
+  makeFlags = [
+    "TARGET_PLATFORM=i686-linux" # the project is 32-bit only for now
+  ];
```

Comment:
upstream supports three different versions through ROMID, and a DEBUG flag for a debug menu. Perhaps for a later PR.

Reply:
Yes exactly, I'll add those later.
There's also a network version in another branch btw, but it's not ready yet.